dask_geopandas/geohash.py

```diff
@@ -0,0 +1,224 @@
+import numpy as np
+import pandas as pd
+from .utils import _calculate_mid_points
+
+
+# Implementation based on deprecated https://pypi.org/project/neathgeohash/#description
+
+
+def _geohash(gdf, precision):
+    """
+    Calculate geohash based on the middle points of the geometry bounds
+    for a given precision
+
+    Parameters
+    ----------
+    gdf : GeoDataFrame
+    precision : int
+        precision of the Geohash
+
+    Returns
+    ---------
+    type : pandas.Series
+        Series containing geohash
+    """
+
+    # Calculate bounds
+    bounds = gdf.bounds.to_numpy()
+    # Calculate mid points based on bounds
+    x_mids, y_mids = _calculate_mid_points(bounds)
+    # Create pairs of x and y midpoints
+    coords = np.array([y_mids, x_mids]).T
+    # Encode coords with Geohash
+    geohash = encode_geohash(coords, precision)
+
+    return pd.Series(geohash, index=gdf.index, name="geohash")
+
+
+def encode_geohash(coords, precision):
+    """
+    Calculate geohash based on coordinates for a
+    given precision
+
+    Parameters
+    ----------
+    coords : array_like of shape (n, 2)
+        array of [x, y] pairs
+    precision : int
+        precision of the Geohash
+
+    Returns
+    ---------
+    geohash: array containing geohashes for each coordinate
+    """
+
+    quantized_coords = _encode_quantize_points(coords)
+    g_uint64 = _encode_into_uint64(quantized_coords)
+    gs_uint8_mat = _encode_base32(g_uint64)
+    geohash = _encode_unicode(gs_uint8_mat, precision)
+
+    return geohash
+
+
+def _encode_quantize_points(coords):
+    """
+    Quantize coordinates by mapping onto
+    unit intervals [0, 1] and multiplying by 2^32.
+
+    Parameters
+    ----------
+    coords : array_like of shape (n, 2)
+        array of [x, y] pairs
+        coordinate pairs
+
+    Returns
+    ---------
+    quantized_coords : array_like
+        quantized coordinate pairs
+    """
+
+    _q = np.array([(2.0 ** 32 / 180, 0), (0, 2.0 ** 32 / (180 * 2))], dtype="float64")
+
+    quantized_coords = coords + np.array((90, 180))
+    quantized_coords = np.dot(quantized_coords, _q)
+    quantized_coords = np.floor(quantized_coords)
+
+    return quantized_coords
+
+
+def _encode_into_uint64(quantized_coords):
+    """
+    Encode quantized coordinates into uint64
+
+    Implementation based on "Geohash in Golang Assembly"
+    blog (https://mmcloughlin.com/posts/geohash-assembly)
+
+    Parameters
+    ----------
+    quantized_mat : array_like
+        coordinate pairs
+
+    Returns
+    ---------
+    array_like of shape (n, 2)
+        coordinate pairs encoded to uint64 values
+        quantized coordinate pairs
+    """
+
+    __s1 = np.array([(1, 0), (0, 2)], dtype="uint64")
+
+    g_uint64 = np.uint64(quantized_coords)
```

Comment:
```suggestion
    g_uint64 = quantized_coords.astype(np.uint64)
```

(also a bit more typical usage, and thus easier to read for most)

Reply:
updated in 72703a4a68f302119ac721c838b504ef58bbfa2d